src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedEmbeddedAttributeSymbol.cs

```diff
@@ -0,0 +1,197 @@
+
+        internal override NamedTypeSymbol GetDeclaredBaseType(ConsList<Symbol> basesBeingResolved) => _baseType;
+
+        internal override ImmutableArray<NamedTypeSymbol> GetDeclaredInterfaces(ConsList<Symbol> basesBeingResolved) => ImmutableArray<NamedTypeSymbol>.Empty;
+        internal override ImmutableArray<Symbol> GetEarlyAttributeDecodingMembers() => GetMembers();
+
+        internal override ImmutableArray<Symbol> GetEarlyAttributeDecodingMembers(string name) => GetMembers(name);
+
+        internal override IEnumerable<FieldSymbol> GetFieldsToEmit() => SpecializedCollections.EmptyEnumerable<FieldSymbol>();
+
+        internal override ImmutableArray<NamedTypeSymbol> GetInterfacesToEmit() => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        internal override IEnumerable<SecurityAttribute> GetSecurityInformation() => null;
+
+        internal override ImmutableArray<NamedTypeSymbol> InterfacesNoUseSiteDiagnostics(ConsList<Symbol> basesBeingResolved = null) => ImmutableArray<NamedTypeSymbol>.Empty;
+
+        internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, ref ArrayBuilder<SynthesizedAttributeData> attributes)
+        {
+            base.AddSynthesizedAttributes(moduleBuilder, ref attributes);
+
+            AddSynthesizedAttribute(
+                ref attributes,
+                moduleBuilder.Compilation.TrySynthesizeAttribute(WellKnownMember.System_Runtime_CompilerServices_CompilerGeneratedAttribute__ctor));
+
+            AddSynthesizedAttribute(
+                ref attributes,
+                moduleBuilder.SynthesizeEmbeddedAttribute());
+        }
+
+        internal sealed class SynthesizedEmbeddedAttributeConstructorSymbol : SynthesizedInstanceConstructor
+        {
+            public SynthesizedEmbeddedAttributeConstructorSymbol(NamedTypeSymbol containingType)
+                : base(containingType)
+            {
+            }
+
+            internal override void GenerateMethodBody(TypeCompilationState compilationState, DiagnosticBag diagnostics)
+            {
+                if (ContainingType.BaseType is MissingMetadataTypeSymbol)
+                {
+                    // System_Attribute is missing. Don't generate anything
+                    return;
+                }
+
+                var factory = new SyntheticBoundNodeFactory(this, this.GetNonNullSyntaxNode(), compilationState, diagnostics);
+                factory.CurrentMethod = this;
+
+                var baseConstructorCall = MethodCompiler.GenerateBaseParameterlessConstructorInitializer(this, diagnostics);
+                if (baseConstructorCall == null)
+                {
+                    // This may happen if Attribute..ctor is not found or is inaccessible
+                    return;
+                }
+
+                var statements = new BoundStatement[]
+                {
+                    factory.ExpressionStatement(baseConstructorCall),
+                    factory.Return(),
+                };
```

Comment:
Minor point: `factory.Block()` has overload with `params` array so these could be inlined below.